### ISSUE TITLE
Firefox: Pressing Up arrow takes you two lines up, if cursor is at beginning of second line of paragraph

### ISSUE THREAD

**Issue by nyanpasu64** [WordPress/gutenberg#34215]
### Description

In a multi-line paragraph, if the cursor is at the beginning of the second line, pressing Up takes you two lines up instead of one. (Gutenberg acts as if the cursor was still within the first line.)

### Step-by-step reproduction instructions

1. Edit a post with any block (quote, paragraph...) followed by a paragraph.
2. Place the cursor at the second line of the following paragraph.
3. Press the Up arrow key.

If the "following paragraph" is a quote instead, pressing Up will highlight the quote block (as if the cursor was on the first line) instead of moving the cursor to the first line.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/913957/130326911-b979f81c-69e3-41a1-acf4-656700ecbf02.mp4

### Environment info

- Version: Wordpress 5.8, built-in editor, Argon and 2021 theme. Also reproduced on https://wordpress.org/gutenberg/.
- Browser: Firefox and Edge
- Device: Laptop with Windows 10 x64

### Pre-checks

- [X] I have searched the existing issues.
- [X] I have tested with all plugins deactivated except Gutenberg.

**Comment by westnz** [comment]
I was able to reproduce this.

**Comment by skorasaurus** [comment]
can confirm this on Gutenberg 13.4; encountered this while troubleshooting https://github.com/WordPress/gutenberg/issues/41416

**Comment by ellatrix** [comment]
Yes, I encountered this while adding e2e tests for Firefox in #53458.

**Comment by Mamaduka** [comment]
Here's my proposed fix - #76954.
